Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1457,8 +1457,12 @@ void HTMLElement::popoverAttributeChanged(const AtomString& value)
         { CSSSelector::PseudoClassClosed, newPopoverState != PopoverState::None }
     });
 
-    if (oldPopoverState != PopoverState::None)
-        hidePopoverInternal(FocusPreviousElement::Yes, FireEvents::No);
+    if (oldPopoverState != PopoverState::None) {
+        auto oldValue = attributeWithoutSynchronization(popoverAttr);
+        hidePopoverInternal(FocusPreviousElement::Yes, FireEvents::Yes);
+        if (!isConnected() || !popoverData() || oldValue != attributeWithoutSynchronization(popoverAttr))
```

Comment:
Why do we skip `ensurePopoverData().setPopoverState(newPopoverState);` in the case of `!isConnected()` ?

Does this work properly?
```
div = document.createElement("div"); // create disconnected div
div.popover = "auto";
div.popover = "manual";
document.body.append(div);
```

You should get the manual popover state after the two popover assignments.